Update CI flow

soopyc opened the pull request:
This PR contains the following changes:

- update various action workflows
- update flake lock
- bumps maximum size for ISO files to 2000M (should be lower than GitHub's 2GiB limit)
- only do ISO splits when the size exceeds 2000M
- change ISO naming scheme
  - it should be clearer by default that ISO files are NixOS ISO files after downloading, especially when working with different distros.
- configure dependabot

soopyc committed:
ci: update action versions

soopyc committed:
ci: split isos only when needed, bump max size and change naming format

soopyc committed:
chore: add gitignore

AdityaGarg8 commented:
You can use dependabot to automatically update actions:

https://github.com/t2linux/T2-Debian-and-Ubuntu-Kernel/blob/Mainline/.github/dependabot.yml

soopyc committed:
ci: configure dependabot

soopyc commented:
Waiting for upstream https://github.com/NixOS/nixos-hardware/pull/1030 to merge before merging this one.

soopyc committed:
flake.lock: Update

Flake lock file updates:

• Updated input 'nixos-hardware':
    'github:NixOS/nixos-hardware/6e253f12b1009053eff5344be5e835f604bb64cd?narHash=sha256-xNbjISJTFailxass4LmdWeV4jNhAlmJPwj46a/GxE6M%3D' (2024-07-02)
  → 'github:NixOS/nixos-hardware/da0aa7b533d49e6319c603e07b46a5690082f65f?narHash=sha256-bwy1rPQSQSCj/TNf1yswHW88nBQYvJQkeScGvOA8pd4%3D' (2024-07-07)
• Updated input 'nixpkgs':
    'github:NixOS/nixpkgs/00d80d13810dbfea8ab4ed1009b09100cca86ba8?narHash=sha256-H3%2BEC5cYuq%2BgQW8y0lSrrDZfH71LB4DAf%2BTDFyvwCNA%3D' (2024-07-01)
  → 'github:NixOS/nixpkgs/9f4128e00b0ae8ec65918efeba59db998750ead6?narHash=sha256-rwz8NJZV%2B387rnWpTYcXaRNvzUSnnF9aHONoJIYmiUQ%3D' (2024-07-03)
